### PR TITLE
JBR-7419 Refactor Vulkan code

### PR DESCRIPTION
**Pull request by YaaZ**
- Separate instance and device-specific function tables
- Avoid using device from global context when possible
- Set up debug logger

### CODE REVIEW

**Review by avu — APPROVED** [review]
LGTM

File: src/java.desktop/share/native/common/java2d/vulkan/VKImage.c [hunk 1]
```diff
@@ -188,14 +181,14 @@ VKImage* VKImage_CreateImageArrayFromSwapChain(VkSwapchainKHR swapchainKhr, VkRe
                 .noImageDealloc = VK_TRUE
         }));
 
-        if (!VKImage_CreateView(&ARRAY_LAST(images))) {
-            ARRAY_APPLY(images, VKImage_dealloc);
+        if (!VKImage_CreateView(logicalDevice, &ARRAY_LAST(images))) {
+            for (uint32_t j = 0; j < ARRAY_SIZE(images); j++) VKImage_dealloc(logicalDevice, &(images[j]));
```
[the remark this answers is not in the document]
Comment (YaaZ) [reply src/java.desktop/share/native/common/java2d/vulkan/VKImage.c:185]: Passing logicalDevice is exactly why I got rid of ARRAY_APPLY. I don't see how we can make general-purpose macro, which would insert array element parameter into arbitrary place. Could you explain your suggestion?

File: src/java.desktop/share/native/common/java2d/vulkan/VKBase.c [hunk 2]
```diff
@@ -804,13 +751,63 @@ jboolean VK_CreateLogicalDevice(jint requestedDevice) {
...
+JNIEXPORT jboolean JNICALL
+Java_sun_java2d_vulkan_VKInstance_initNative(JNIEnv *env, jclass wlge, jlong nativePtr, jboolean verb, jint requestedDevice) {
+    verbose = verb;
+    PFN_vkGetInstanceProcAddr vkGetInstanceProcAddr = vulkanLibOpen();
+    if (vkGetInstanceProcAddr == NULL) {
+        return JNI_FALSE;
+    }
+    geInstance = (VKGraphicsEnvironment*)malloc(sizeof(VKGraphicsEnvironment));
+    if (geInstance == NULL) {
+        J2dRlsTrace(J2D_TRACE_ERROR, "Vulkan: Cannot allocate VKGraphicsEnvironment\n")
```
[the remark this answers is not in the document]
Comment (YaaZ) [reply src/java.desktop/share/native/common/java2d/vulkan/VKBase.c:784]: I did it for conformity with other messages, but I'll better refactor them, I guess

File: src/java.desktop/share/native/common/java2d/vulkan/VKImage.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (avu) [reply src/java.desktop/share/native/common/java2d/vulkan/VKImage.c:185]: We can use variadic macro here (C99 feature) https://gcc.gnu.org/onlinedocs/cpp/Variadic-Macros.html

**Pull request merged**